code/modules/mob/living/brain/brain_item.dm

```diff
@@ -249,13 +269,23 @@
 	var/obj/item/organ/brain/replacement_brain = replacement
 	if(!istype(replacement_brain))
 		return
-	for(var/obj/item/skillchip/skill_chip in src)
-		if(skill_chip in skillchips)
-			if(owner)
-				skill_chip.on_removal(owner)
-			LAZYREMOVE(skillchips, skill_chip)
-			LAZYADD(replacement_brain.skillchips, skill_chip) //No need to call on_apply here, since it will be inserted in organ replacement soon.
-		skill_chip.forceMove(replacement)
+
+	// Check through all our skillchips, remove them from this brain, add them to the replacement brain.
+	for(var/chip in skillchips)
+		var/obj/item/skillchip/skillchip = chip
+
+		// We're technically doing a little hackery here by bypassing the procs, but I'm the one who wrote them
+		// and when you know the rules, you can break the rules.
+
+		// Technically the owning mob is the same. We don't need to activate or deactivate the skillchips.
+		// All the skillchips themselves care about is what brain they're in.
+		// Because the new brain will ultimately be owned by the same body, we can safely leave skillchip logic alone.
+
+		// Directly change the new holding_brain.
+		skillchip.holding_brain = replacement_brain
+
+		// Directly add them to the skillchip list in the new brain.
+		LAZYADD(replacement_brain.skillchips, skillchip)
```

Comment:
This might be bit dangerous if we add chips depending on specific brain subtype.

Reply:
Sorted this out.

If the brain subtypes are different, a failsafe is engaged that attempts to deactivate all active skillchips. The player gets a feedback message about this event.